### PR TITLE
dash action path, delete routes file

### PR DESCRIPTION
**Pull request by MasaKni**

### CODE REVIEW

File: tests/TestCase/Command/RoutesCommandTest.php [hunk 1]
```diff
@@ -189,6 +189,6 @@ public function tearDown(): void
         parent::tearDown();
 
         // remove routes files
-        \array_map('unlink', \glob(CONFIG . 'routes*.php'));
+        \array_map('unlink', \glob(CONFIG . '*routes*.php'));
```
Comment (segy) [inline tests/TestCase/Command/RoutesCommandTest.php]: why this change? was the glob not working without the leading `*`?

File: src/Attribute/Connect.php [hunk 2]
```diff
@@ -25,7 +27,7 @@ public function __construct(
 
     public function phpCode(): string
     {
-        return '$builder->connect(' . $this->varExport($this->getUri()) . ', ' . $this->varExport($this->getOptions()) . ');';
+        return '$builder->connect(' . $this->varExport($this->getUri()) . ', ' . $this->varExport(['controller' => $this->controller, 'action' => $this->action]) . ', ' . $this->varExport($this->getOptions()) . ');';
```
Comment (segy) [inline src/Attribute/Connect.php]: i'm not convinced about this. please add tests with special options like `_ext` or `_method` (see https://book.cakephp.org/5/en/development/routing.html#reserved-route-elements)

**Review by segy — CHANGES_REQUESTED** [review]

File: tests/TestCase/Command/RoutesCommandTest.php (hunk 1 above)
Comment (MasaKni) [reply tests/TestCase/Command/RoutesCommandTest.php]: yes, the compiled_routes doesn't get deleted

File: src/Attribute/Connect.php [hunk 3]
```diff
@@ -35,14 +41,22 @@ public function setAction(string $action): void
 
     protected function getUri(): string
     {
-        return (\str_starts_with($this->uri, '/') ? '' : '/' . \strtolower($this->controller) . '/') . ($this->uri ?: \strtolower($this->action));
+        return (\str_starts_with($this->uri, '/') ? '' : '/' . Inflector::dasherize($this->controller) . '/') . ($this->uri ?: Inflector::dasherize($this->action));
```
Comment (segy) [inline src/Attribute/Connect.php:47]: does `dasherize` also lowercase string?

File: src/Attribute/Connect.php [hunk 4]
```diff
@@ -8,14 +8,20 @@
...
     protected string $action;
 
-    /** @param mixed[] $options */
+    /**
+     * @param mixed[] $options
+     * @param mixed[] $defaults
+     */
     public function __construct(
         protected string $uri = '',
+        protected array $defaults = [],
```
Comment (segy) [inline src/Attribute/Connect.php:27]: what are `defaults`? is it a param of the `connect` method? i see no comment here so plese try to explain, when you do a change like this.

File: src/Command/RoutesCommand.php [hunk 5]
```diff
@@ -71,6 +71,7 @@ public function execute(Arguments $args, ConsoleIo $io): void
             $plugin = $plugin ? ", ['plugin' => '{$plugin}']" : '';
 
             $phpCode .= "\n\$routes->scope('{$scope}'{$plugin}, function (\\Cake\\Routing\\RouteBuilder \$builder) {\n";
+            $phpCode .= "    \$builder->setExtensions(['json', 'xml']);\n";
```
Comment (segy) [inline src/Command/RoutesCommand.php]: i don't think this is needed to do globally- you can set this manually in the routes file, if you need to use them.

File: tests/TestCase/Command/RoutesCommandTest.php (hunk 1 above)
Comment (segy) [reply tests/TestCase/Command/RoutesCommandTest.php]: isn't the name `routes_compiled`?

**Review by segy — CHANGES_REQUESTED** [review]

File: src/Attribute/Connect.php (hunk 3 above)
Comment (MasaKni) [reply src/Attribute/Connect.php:47]: yes

File: tests/TestCase/Command/RoutesCommandTest.php (hunk 1 above)
Comment (MasaKni) [reply tests/TestCase/Command/RoutesCommandTest.php]: there is a file core_routes, i renamed it to routes_core
